File: Sorting/InsertionSort.cpp
#include <iostream>
using namespace std;

void insertionSort(int arr[], int n) {
    for (int i = 1; i < n; i++) {
        int temp = arr[i];
        int j = i - 1;
        while (j >= 0 && arr[j] > temp) {
            arr[j + 1] = arr[j];
            j--;
        }
        arr[j + 1] = temp;
    }
}

int main() {
    int arr[10] = {64, 25, 12, 22, 11, 90, 45, 78, 34, 23};
    insertionSort(arr, 10);
    cout << "Sorted array: \n";
    for (int i = 0; i < 10; i++) {
        cout << arr[i] << " ";
    }
    cout << endl;
    return 0;
}

File: Sorting/InsertionSort.py
nums = [3,321,412,431,431,5421,4,21,44]
n = len(nums)
def insertionSort(nums):
    for i in range(1, n):
        key = nums[i]
        j = i-1
        while j>=0 and nums[j]>key:
            nums[j+1] = nums[j]
            j -= 1
        nums[j+1] = key
    return nums
print(insertionSort(nums))